discord/flags.py

```diff
@@ -410,14 +434,14 @@ def all(cls):
         return self
 
     @classmethod
-    def none(cls):
+    def none(cls: Type[I]) -> I:
```

Comment:
I don't think this pattern actually needs to be generic.

```py
class C:
    @classmethod
    def foo(cls) -> 'C':
        return cls()
```

The following works and checks without any extra work. Admittedly, if for whatever reason these types would be inherited then the whole thing kind of falls apart. I'm not sure why the typing module doesn't have a `typing.Self` and ` typing.SelfType` honestly considering how common these two are but that's an aside.

It can stay as-is, I was just trying to understand why the generic was there.

However, note that rather than specifying the 3 different bound `TypeVar` you can just use the `BaseFlag` bound `TypeVar` and the type checker will accept it.

```py
T = TypeVar('T', bound='A')

class A:
    pass

class B(A):
    @classmethod
    def foo(cls: Type[T]) -> T:
        return cls()

b = B.foo() # resolved type: B
```

This is because all flags are required to have `BaseFlags` as its bottom type.